cases/util/validateURL.js

```diff
@@ -0,0 +1,9 @@
+export const isURLvalid = async (str) => {
+    var pattern = new RegExp('^(https?:\\/\\/)?'+ // protocol
+      '((([a-z\\d]([a-z\\d-]*[a-z\\d])*)\\.)+[a-z]{2,}|'+ // domain name
+      '((\\d{1,3}\\.){3}\\d{1,3}))'+ // OR ip (v4) address
+      '(\\:\\d+)?(\\/[-a-z\\d%_.~+]*)*'+ // port and path
+      '(\\?[;&a-z\\d%_.~+=-]*)?'+ // query string
+      '(\\#[-a-z\\d_]*)?$','i'); // fragment locator
```

Comment:
I don't think the transfer server URL would have a fragment locator since its not a web page, right?

Reply:
would it be easier to just do `new URL(str)` and let that throw an error if its not a valid url?

Reply:
I thought about making this as broad as possible to be reused in the future in other URL validations.

Reply:
are there any URLs that wouldn't pass a `new URL()` test?  I'd rather be too strict and loosen it than too loose and allow broken things (and confusing code)